pkg/frontend/admin_openshiftcluster_etcdcertificaterenew_test.go

```diff
@@ -0,0 +1,508 @@
+			mocks: func(tt *test, k *mock_adminactions.MockKubeActions) {
+				k.EXPECT().
+					KubeGet(gomock.Any(), "ClusterVersion.config.openshift.io", "", "version").MaxTimes(1).
+					Return(encodeClusterVersion(t, tt.version), nil)
+				k.EXPECT().
+					KubeGet(gomock.Any(), "etcd.operator.openshift.io", "", "cluster").AnyTimes().
+					Return(encodeEtcdOperatorController(t, tt.etcdoperator), nil)
+				k.EXPECT().
+					KubeGet(gomock.Any(), "ClusterOperator.config.openshift.io", "", "etcd").AnyTimes().
+					Return(encodeEtcdOperator(t, tt.etcdCO), nil)
+				k.EXPECT().
+					KubeGet(gomock.Any(), "Secret", namespaceEtcds, gomock.Any()).AnyTimes().
+					Return(CreateCertSecret(t, tt.notBefore, tt.notAfter), nil)
+			},
+			wantStatusCode: http.StatusInternalServerError,
+			wantError:      "500: InternalServerError: : secret etcd-peer-cluster-aro-master-0 is already expired, quitting.",
+		},
+		{
+			name:       "delete etcd secrets",
+			resourceID: fmt.Sprintf("/subscriptions/%s/resourcegroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openShiftClusters/resourceName", mockSubID),
+			version: &configv1.ClusterVersion{
+				Status: configv1.ClusterVersionStatus{
+					History: []configv1.UpdateHistory{
+						{
+							State:   configv1.CompletedUpdate,
+							Version: "4.8.11",
+						},
+					},
+				},
+			},
+			etcdoperator: &operatorv1.Etcd{
+				Status: operatorv1.EtcdStatus{
+					StaticPodOperatorStatus: operatorv1.StaticPodOperatorStatus{
+						OperatorStatus: operatorv1.OperatorStatus{
+							Conditions: []operatorv1.OperatorCondition{
+								{
+									Type:   "EtcdCertSignerControllerDegraded",
+									Status: operatorv1.ConditionFalse,
+								},
+							},
+						},
+						LatestAvailableRevision: 1,
+						NodeStatuses: []operatorv1.NodeStatus{
+							{
+								NodeName:        "master-0",
+								CurrentRevision: 1,
+							},
+						},
+					},
+				},
+			},
+			etcdoperatorRevisied: &operatorv1.Etcd{
+				Status: operatorv1.EtcdStatus{
+					StaticPodOperatorStatus: operatorv1.StaticPodOperatorStatus{
+						OperatorStatus: operatorv1.OperatorStatus{
+							Conditions: []operatorv1.OperatorCondition{
+								{
+									Type:   "EtcdCertSignerControllerDegraded",
+									Status: operatorv1.ConditionFalse,
+								},
+							},
+						},
+						LatestAvailableRevision: 2,
+						NodeStatuses: []operatorv1.NodeStatus{
+							{
+								NodeName:        "master-0",
+								CurrentRevision: 2,
+							},
+						},
+					},
+				},
+			},
+			etcdCO: &configv1.ClusterOperator{
+				Status: configv1.ClusterOperatorStatus{
+					Conditions: []configv1.ClusterOperatorStatusCondition{
+						{
+							Type:   configv1.OperatorDegraded,
+							Status: configv1.ConditionFalse,
+						},
+					},
+				},
+			},
+			notBefore: time.Now().AddDate(-2, -8, 0),
+			notAfter:  time.Now().Add((1 * time.Hour)),
+			mocks: func(tt *test, k *mock_adminactions.MockKubeActions) {
+				k.EXPECT().
```

Comment:
We should be testing the specific number of calls, since this ought to be deterministic?